src/config.rs

```diff
@@ -878,6 +881,12 @@ impl Config {
             .map(|p| p.as_path())
     }
 
+    pub fn shell(&self) -> Option<&Path> {
+        self.shell
+            .as_ref()
+            .map(|p| p.as_path())
```

Comment:
How is this different than the config_path case which uses `.map(|p| p.as_path())` as well? (I'm new to Rust, so forgive the unrelated question)